packages/astro/src/core/app/index.ts

```diff
@@ -131,8 +134,31 @@ export class App {
 		const url = new URL(request.url);
 		// ignore requests matching public assets
 		if (this.#manifest.assets.has(url.pathname)) return undefined;
-		const pathname = prependForwardSlash(this.removeBase(url.pathname));
-		const routeData = matchRoute(pathname, this.#manifestData);
+		let pathname;
+		if (this.#manifest.i18n && this.#manifest.i18n.routingStrategy === 'domain') {
+			const host = request.headers.get('X-Forwarded-Host');
```

Comment:
Is `X-Forwarded-Host` the only header to check here? I think that `X-Forwarded-Host` only occurs when there is a proxy server sitting in front. Should we fallback to `Host` itself?

Reply:
Yeah we really should. I will update. the logic